src/pybind/mgr/dashboard/services/cluster.py

```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
+from enum import Enum
+
+from .. import mgr
+
+
+class ClusterModel:
+
+    class Status(Enum):
+        INSTALLED = 0
+        POSTINSTALLED = 1
+
+    status: Status
+
+    def __init__(self, status=Status.INSTALLED.name):
+        self.status = status
+
+    def dict(self):
+        return {'status': self.status}
+
+    def to_db(self):
+        mgr.set_store('cluster/status', self.status)
+
+    @classmethod
+    def from_db(cls):
+        return cls(status=mgr.get_store('cluster/status', cls.Status.INSTALLED.name))
```

Comment:
This is a tricky thing. `ClusterModel.status` is an Enum, but `Status.INSTALLED.name` is a string (`mypy` should be catching this error) so you need to decide where to transform from string (or number) to Enum:
```suggestion
    def __init__(self, status: str=Status.INSTALLED.name):
        self.status = Status[status] # from name to Enum

    def dict(self):
        return {'status': self.status.name}

    def to_db(self):
        mgr.set_store('cluster/status', self.status.value)

    @classmethod
    def from_db(cls):
        return cls(
          status=Status(
               mgr.get_store('cluster/status', cls.Status.INSTALLED.name)
             ).name # from value to Enum and then name
           )
```
Or
```suggestion
    def __init__(self, status: Status=Status.INSTALLED):
        self.status = status

    def dict(self):
        return {'status': self.status.name}

    def to_db(self):
        mgr.set_store('cluster/status', self.status.value)

    @classmethod
    def from_db(cls):
        return cls(
          status=Status(
            mgr.get_store('cluster/status', cls.Status.INSTALLED.name)
            )
          )
```

Reply:
Done.